src/core/shader.js

```diff
@@ -50,29 +50,27 @@ Shader.prototype = {
    * Called during shader initialization and is only run once.
    */
   init: function (data) {
-    this.attributes = this.initVariables(data, 'attribute');
-    this.uniforms = this.initVariables(data, 'uniform');
+    this.uniforms = this.initUniforms();
     this.material = new (this.raw ? THREE.RawShaderMaterial : THREE.ShaderMaterial)({
-      // attributes: this.attributes,
       uniforms: this.uniforms,
       vertexShader: this.vertexShader,
       fragmentShader: this.fragmentShader
     });
     return this.material;
   },
 
-  initVariables: function (data, type) {
+  initUniforms: function () {
     var key;
     var schema = this.schema;
     var variables = {};
     var varType;
 
     for (key in schema) {
-      if (schema[key].is !== type) { continue; }
+      if (schema[key].is !== 'uniform') { continue; }
```

Comment:
Should 'is' property be removed from schema and all shaders updated accordingly?

Reply:
I did contemplate that as well, but I think it's safer to leave it. It can also serve some purpose. Say for example you rename or deprecate a uniform on your shader. Now you can add a property to the schema under the old name, but not mark it as a uniform. By overloading the `update` method you can then print a warning message and delegate to the old behaviour without it ending up as a uniform.